apps/web/src/design-system/sidebar/Sidebar.tsx

```diff
@@ -0,0 +1,156 @@
+import { ReactNode, useEffect } from 'react';
+import { ActionIcon, createStyles, Drawer, Loader, MantineTheme, Stack } from '@mantine/core';
+import styled from '@emotion/styled';
+
+import { colors, shadows } from '../config';
+import { ArrowLeft, Close } from '../icons';
+import { When } from '../../components/utils/When';
+
+const HeaderHolder = styled.div`
+  display: flex;
+  flex-wrap: nowrap;
+  gap: 12px;
+  margin: 24px;
+  margin-bottom: 0;
+`;
+
+const BodyHolder = styled.div`
+  display: flex;
+  flex-direction: column;
+  overflow-x: hidden;
+  overflow-y: auto;
+  margin: 0 24px;
+  gap: 24px;
+`;
+
+const FooterHolder = styled.div`
+  display: flex;
+  flex-wrap: nowrap;
+  gap: 6px;
+  margin: 24px;
+  margin-top: 0;
+  margin-top: auto;
+`;
+
+const COLLAPSED_WIDTH = 480;
+const HEADER_HEIGHT = 65;
+const NAVIGATION_WIDTH = 300;
+const PAGE_MARGIN = 30;
+const INTEGRATION_SETTING_TOP = HEADER_HEIGHT + PAGE_MARGIN;
+
+const useDrawerStyles = createStyles((theme: MantineTheme) => {
+  return {
+    root: {
+      position: 'absolute',
+      zIndex: 1,
+    },
+    drawer: {
+      position: 'fixed',
+      top: `${INTEGRATION_SETTING_TOP}px`,
+      right: `${PAGE_MARGIN}px`,
+      bottom: 0,
+      backgroundColor: theme.colorScheme === 'dark' ? colors.B17 : colors.white,
+      borderRadius: 7,
+      boxShadow: shadows.dark,
+    },
+    body: {
+      height: '100%',
+    },
+  };
+});
+
+const Form = styled.form`
+  height: 100%;
+  overflow: hidden;
+  display: flex;
+  flex-direction: column;
+  gap: 24px;
+`;
+
+export const Sidebar = ({
+  customFooter,
+  customHeader,
+  children,
+  isOpened,
+  isExpanded = false,
+  isLoading = false,
+  onClose,
+  onBack,
+  onSubmit,
+}: {
+  customHeader?: ReactNode;
+  customFooter?: ReactNode;
+  children: ReactNode;
+  isOpened: boolean;
+  isExpanded?: boolean;
+  isLoading?: boolean;
+  onClose: () => void;
+  onBack?: () => void;
+  onSubmit?: React.FormEventHandler<HTMLFormElement>;
+}) => {
+  const { classes: drawerClasses } = useDrawerStyles();
+
+  useEffect(() => {
+    const handleKeyDown = (e) => {
+      if (e.key === 'Escape') {
+        onClose();
+      }
+    };
```

Comment:
In https://github.com/novuhq/novu/blob/8e71c9b09144b5f950c6d8317e2339964051bd93/apps/web/src/pages/integrations/IntegrationsStoreModal.tsx#L102 we also have a `Esc` keybinding. Could it be a good idea to centralise keybindings in one place in a way we could provide a desired function to execute if pressed? Improvement for the future for sure.

Reply:
Do you mean encapsulate it in the hook? yes, good suggestion will do that :)